From d3d46719b5648b824370f676fe189df1fb32ad98 Mon Sep 17 00:00:00 2001
From: Rajat Soni <rajson@qti.qualcomm.com>
Date: Wed, 23 Jul 2025 12:55:43 +0530
Subject: [PATCH] QSDK: Makefile: Update hostapd.uc and wpa_supplicant.uc files
 location

Take hostapd.uc and wpa_supplicant.uc files from wlan-hostapd
component if files are available.

Change-Id: I07675248c2f9cd29896102459ebfb44f7fd6591d
Signed-off-by: Rajat Soni <rajson@qti.qualcomm.com>
---
 package/network/services/hostapd/Makefile | 17 +++++++++++++++++
 1 file changed, 17 insertions(+)

diff --git a/package/network/services/hostapd/Makefile b/package/network/services/hostapd/Makefile
index ea78cb7a322f..3184af9b4b3e 100644
--- a/package/network/services/hostapd/Makefile
+++ b/package/network/services/hostapd/Makefile
@@ -762,16 +762,33 @@ Package/hostapd-wolfssl/conffiles = $(Package/hostapd-full/conffiles)
 Package/hostapd-mbedtls/conffiles = $(Package/hostapd-full/conffiles)
 endif
 
+#Select hostapd.uc file from external hostapd dir if available
+ifneq ("$(wildcard $(FILE_DIR_HOSTAPD)/files/hostapd.uc)","")
+define Install/hostapd
+	$(INSTALL_DIR) $(1)/usr/sbin $(1)/usr/share/hostap
+	$(INSTALL_DATA) $(FILE_DIR_HOSTAPD)/files/hostapd.uc $(1)/usr/share/hostap/
+	$(if $(findstring full,$(CONFIG_VARIANT)),$(Install/hostapd/full))
+endef
+else
 define Install/hostapd
 	$(INSTALL_DIR) $(1)/usr/sbin $(1)/usr/share/hostap
 	$(INSTALL_DATA) ./files/hostapd.uc $(1)/usr/share/hostap/
 	$(if $(findstring full,$(CONFIG_VARIANT)),$(Install/hostapd/full))
 endef
+endif
 
+#Select wpa_supplicant.uc file from external hostapd dir if available
+ifneq ("$(wildcard $(FILE_DIR_HOSTAPD)/files/wpa_supplicant.uc)","")
+define Install/supplicant
+	$(INSTALL_DIR) $(1)/usr/sbin $(1)/usr/share/hostap
+	$(INSTALL_DATA) $(FILE_DIR_HOSTAPD)/files/wpa_supplicant.uc $(1)/usr/share/hostap/
+endef
+else
 define Install/supplicant
 	$(INSTALL_DIR) $(1)/usr/sbin $(1)/usr/share/hostap
 	$(INSTALL_DATA) ./files/wpa_supplicant.uc $(1)/usr/share/hostap/
 endef
+endif
 
 define Package/hostapd-common/install
 	$(INSTALL_DIR) $(1)/etc/capabilities $(1)/etc/rc.button $(1)/etc/hotplug.d/ieee80211 $(1)/etc/init.d $(1)/lib/netifd  $(1)/usr/share/acl.d $(1)/usr/share/hostap
-- 
2.34.1

